### PR TITLE
dns-over-https: init at 2.3.7

### PR DESCRIPTION
**Pull request by ghost**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/dn/dns-over-https/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  nix-update-script,
+}:
+buildGoModule rec {
+  pname = "dns-over-https";
+  version = "2.3.7";
+
+  src = fetchFromGitHub {
+    owner = "m13253";
+    repo = pname;
+    rev = "v${version}";
```
[the remark this answers is not in the document]
Comment (cry0ice) [reply pkgs/by-name/dn/dns-over-https/package.nix]: Solved

File: pkgs/by-name/dn/dns-over-https/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  ];
+
+  subPackages = [
+    "doh-client"
+    "doh-server"
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
```
Comment (drupol) [inline pkgs/by-name/dn/dns-over-https/package.nix:32]: Missing `meta.mainProgram`.

File: pkgs/by-name/dn/dns-over-https/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  buildGoModule,
+  nix-update-script,
+}:
+buildGoModule rec {
+  pname = "dns-over-https";
+  version = "2.3.7";
+
+  src = fetchFromGitHub {
+    owner = "m13253";
+    repo = pname;
```
Comment (drupol) [inline pkgs/by-name/dn/dns-over-https/package.nix:13]: Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

File: pkgs/by-name/dn/dns-over-https/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    tag = "v${version}";
```
Comment (drupol) [inline pkgs/by-name/dn/dns-over-https/package.nix:14]: Are you sure that the tag has a `v` prefix ?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/dn/dns-over-https/package.nix (hunk 2 above)
Comment (cry0ice) [reply pkgs/by-name/dn/dns-over-https/package.nix:32]: It contains two programs. I don't know how to set it.

File: pkgs/by-name/dn/dns-over-https/package.nix (hunk 4 above)
Comment (cry0ice) [reply pkgs/by-name/dn/dns-over-https/package.nix:14]: Yes, I am sure.

Comment (drupol) [reply pkgs/by-name/dn/dns-over-https/package.nix:14]: Fair enough.

File: pkgs/by-name/dn/dns-over-https/package.nix (hunk 3 above)
Comment (cry0ice) [reply pkgs/by-name/dn/dns-over-https/package.nix]: Resolved

**Comment by drupol** [comment]
@ofborg build dns-over-https

**Review by drupol — APPROVED** [review]

**Pull request merged**